gazebo_plugins/test/test_gazebo_ros_force.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/test/ServerFixture.hh>
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/wrench.hpp>
+
+#include <memory>
+
+#define tol 10e-10
+
+class GazeboRosForceTest : public gazebo::ServerFixture
+{
+};
+
+TEST_F(GazeboRosForceTest, ApplyForceTorque)
+{
+  // Load test world and start paused
+  this->Load("worlds/gazebo_ros_force.world", true);
+
+  // World
+  auto world = gazebo::physics::get_world();
+  ASSERT_NE(nullptr, world);
+
+  // Box
+  auto box = world->ModelByName("box");
+  ASSERT_NE(nullptr, box);
+
+  // Check plugin was loaded
+  world->Step(100);
+  EXPECT_EQ(1u, box->GetPluginCount());
+
+  // Check box is at world origin
+  EXPECT_NEAR(0.0, box->WorldPose().Pos().X(), tol);
+  EXPECT_NEAR(0.0, box->WorldPose().Rot().Z(), tol);
+
+  // Create ROS publisher
+  auto node = std::make_shared<rclcpp::Node>("gazebo_ros_force_test");
+  ASSERT_NE(nullptr, node);
+
+  auto pub = node->create_publisher<geometry_msgs::msg::Wrench>("test/force_test");
+
+  // Apply force on X
+  auto msg = geometry_msgs::msg::Wrench();
+  msg.force.x = 10.0;
+  msg.torque.z = 10.0;
+  pub->publish(msg);
+
+  // Wait until box moves
+  unsigned int sleep = 0;
+  unsigned int max_sleep = 30;
+  while (sleep < max_sleep && abs(box->WorldPose().Pos().X()) < tol) {
```

Comment:
Perhaps there should be a larger tolerance for seeing that it moved. 10e-10 could just be the initial rounding error

Reply:
Good idea, increased in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/784/commits/0cc0e06f4aaca8b9cabd025dde907402afc9fc7a